interface/src/app/download/download.spec.js

```diff
@@ -2,21 +2,113 @@
  * Unit tests.
  */
 
-describe('DownloadCtrl', function() {
-  describe('isCurrentUrl', function() {
-    var DownloadCtrl, $location, $scope;
-
-    beforeEach(module('adage'));
-
-    beforeEach(inject(function($controller, _$location_, $rootScope) {
-      $location = _$location_;
-      $scope = $rootScope.$new();
-      DownloadCtrl = $controller('DownloadCtrl',
-                                 {$location: $location, $scope: $scope});
-    }));
-
-    it('validates DownloadCtrl controller', inject(function() {
-      expect(DownloadCtrl).toBeTruthy();
-    }));
+// Test the generic controller properties (without server calls):
+describe('DownloadCtrl w/o server calls', function() {
+  beforeEach(module('adage.download'));
+
+  var ctrl, mockService;
+  // Create a mocked "AnnotationTypes" service, which overwrites the service
+  // in download.js (because it is loaded after "adage.download" module).
+  beforeEach(module(function($provide) {
+    mockService = {
+      get: function() { }
+    };
+    $provide.value('AnnotationTypes', mockService);
+  }));
+
+  beforeEach(inject(function($controller) {
+    ctrl = $controller('DownloadCtrl');
+  }));
+
+  it('should be defined', function() {
+    expect(ctrl).toBeDefined();
```

Comment:
Do we also want `expect(ctrl).toBeTruthy();` somewhere? That would replace the dummy test that was removed above. Although it doesn't do much, it does confirm that the code for that module compiles. It is nice to have a separate test that can quickly pinpoint a compile failure.

Unless maybe I'm missing something, I don't think the `.toBeDefined()` test here will cover that.